(Documentation) Improve README and add some things to make an initial deploy easier

fvictorio opened the pull request:
Besides updating the README with the instructions to run the app locally and the tests, this PR:

- Adds a `server-config-private.example.js`.
- Adds `ganache-cli` as a dev dependency of the project, so that it doesn't relies on globally installed apps.

garatortiz commented:
Maybe we should indicate that to test it locally, you must complete the **ZIP** with `11111` and the **ADDRESS** with `deliverable` or any of the other options in Lob's page (https://lob.com/docs#us-verification-test-environment)

phahulin commented:
@fvictorio I've made some additions to the README (fixed few typos, removed mentions of contract-output from "How to deploy to a real network", added info about ERC780 integration), please take a look.
If this is ready to merge, please remove "In progress" label

fvictorio commented:
@phahulin This should be ready now. I added a Table of contents to the readme, fixed a problem with Travis and removed some files that were added accidentally.
